src/math_tools/mod.rs

```diff
@@ -132,4 +149,14 @@ mod tests {
         res = compute_bernoulli(10, 1, 1.0_f64 / 6.0_f64);
         assert_eq!(res - 0.324 < 0.001, true);
     }
+
+    #[test]
+    fn test_signed_adder() {
+        let res: usize = super::safe_add_signed_unsigned(5, 2);
+        assert_eq!(res, 7);
+        let res: usize = super::safe_add_signed_unsigned(5, -2);
+        assert_eq!(res, 3);
+        let res: usize = super::safe_add_signed_unsigned(5, -10);
+        assert_eq!(res, 0);
+    }
```

Comment:
What happen for -5 and 0 ?

Reply:
The first parameter is a usize, it can't be -5, the code won't compile